### DISCUSSION TITLE
How to clear the entire cache storage?

### DISCUSSION THREAD

**Discussion by shamxeed** [TanStack/query#3280] [Q&A] [answered]
Hi! Thanks for always being there!! How can I clear the entire react-query cache data for example if the user logs out?

What I tried: 

```jsx
const handleLogout = async () => {
   await secureStorage.deleteItemAsync('token')
   await asyncStorage.removeItem('REACT_QUERY_OFFLINE_CACHE')
   ....
}
```

But that doesn't seem to work, please how can I archive that?

**Comment by TkDodo** [comment, marked as answer]
not sure which "cache storage" you are talking about. The react-query in-memory cache can be cleared with `queryClient.clear()`

other than that, if you are using any of the storage plugins - if you clear the queryCache, it should also sync that to external storages.

**Comment by shamxeed** [comment]
Yeah, I'm using a storage plugin, so is there a way to clear the entire storage at once?

**Reply by TkDodo** [reply]
like I said, if you use `queryClient.clear()`, it will clear the cache, and then the empty cache will be synced to the async storage, thus clearing it.

**Reply by mijash-he** [reply]
@TkDodo Is there any time latency from cache to async storage and async storage to cache?
I use `queryClient.clear()` on user logout, and the data in async storage is still left non-cleared.
And if I run `queryClient.clear()` after 1000ms, I can see it's cleared. Note that it's not cleared with 100ms timeout.

The network tab says no http request to the server.


### just `queryClient.clear()`

https://user-images.githubusercontent.com/112013707/224854085-236c84d2-300d-4e5b-80b0-ab75a07b5869.mov


### `queryClient.clear()` with timeout 1000ms

https://user-images.githubusercontent.com/112013707/224854112-09199a93-dd29-4733-93aa-e5c582de9e94.mov

**Reply by TkDodo** [reply]
yes the persisters have a `throttleTime` option: https://tanstack.com/query/v4/docs/react/plugins/createAsyncStoragePersister#createasyncstoragepersister

It defaults to 1000ms

**Reply by mijash-he** [reply]
I see, thank you, @TkDodo !
